src/Temporalio/Bridge/Interop/Interop.cs

```diff
@@ -0,0 +1,408 @@
+using System;
+using System.Runtime.InteropServices;
+
+namespace Temporalio.Bridge.Interop
+{
+    internal enum RpcService
+    {
+        Workflow = 1,
+        Operator,
+        Test,
+        Health,
+    }
+
+    internal partial struct CancellationToken { }
+
+    internal partial struct Client { }
+
+    internal partial struct EphemeralServer { }
+
+    internal partial struct Runtime { }
+
+    internal unsafe partial struct ByteArrayRef
+    {
+        [NativeTypeName("const uint8_t *")]
+        public byte* data;
+
+        [NativeTypeName("size_t")]
+        public UIntPtr size;
+    }
+
+    internal partial struct ClientTlsOptions
+    {
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef server_root_ca_cert;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef domain;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef client_cert;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef client_private_key;
+    }
+
+    internal partial struct ClientRetryOptions
+    {
+        [NativeTypeName("uint64_t")]
+        public ulong initial_interval_millis;
+
+        public double randomization_factor;
+
+        public double multiplier;
+
+        [NativeTypeName("uint64_t")]
+        public ulong max_interval_millis;
+
+        [NativeTypeName("uint64_t")]
+        public ulong max_elapsed_time_millis;
+
+        [NativeTypeName("uintptr_t")]
+        public UIntPtr max_retries;
+    }
+
+    internal unsafe partial struct ClientOptions
+    {
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef target_url;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef client_name;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef client_version;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef metadata;
+
+        [NativeTypeName("struct ByteArrayRef")]
+        public ByteArrayRef identity;
+
+        [NativeTypeName("const struct ClientTlsOptions *")]
+        public ClientTlsOptions* tls_options;
+
+        [NativeTypeName("const struct ClientRetryOptions *")]
+        public ClientRetryOptions* retry_options;
+    }
+
+    internal unsafe partial struct ByteArray
+    {
+        [NativeTypeName("const uint8_t *")]
+        public byte* data;
+
+        [NativeTypeName("size_t")]
+        public UIntPtr size;
+
+        [NativeTypeName("size_t")]
+        public UIntPtr cap;
+
+        [NativeTypeName("bool")]
+        public byte disable_free;
```

Comment:
This is auto-generated from ClangSharp. https://github.com/dotnet/ClangSharp/issues/350#issuecomment-1155385609 mentions that `bool` is `byte` by intention here. After doing some digging (e.g. [this SO answer](https://stackoverflow.com/a/53925256), [this SO answer](https://stackoverflow.com/a/32115697), and [this SO answer](https://stackoverflow.com/a/20037067) show that `bool` is "non-blittable" and therefore unsafe for certain interop uses. So that's why the ClangSharp project did this.

Note btw that this specific field is never used on the C# side, it's for Rust to know not to free the space (it'd be used for static byte arrays).